app/templates/components/goodcity/offers-search-overlay.hbs

```diff
@@ -0,0 +1,58 @@
+<div class="cmp orders search-overlay">
+  {{#popup-overlay open=open}}
+    <div class="main-container">
+      <div class="row search-field">
+        <div class="icon-holder small-1 medium-1 columns">
+          <i {{action "closeOverlay"}} class="back_icon" aria-hidden="true">{{fa-icon 'times' size='lg'}}</i>
+        </div>
+
+        <div class="input-holder small-9 medium-9 columns">
+          {{focus-textfield
+            name="searchText"
+            id=uuid
+            placeholder=(t "search_min")
+            value=searchText }}
+
+          {{#if hasSearchText}}
+            <i class="pinned-right" {{action "clearSearch"}}>{{fa-icon 'times-circle' size='lg'}}</i>
+          {{/if}}
+        </div>
+        {{#if isMobileApp}}
+          <div class="small-2 medium-2 columns scan-button scanner">
+            {{scan-barcode-button returnScannedText=(action "setScannedSearchText")}}
```

Comment:
Callback is a huge improvement over the navigation <3

maybe I suggest renaming `returnScannedText` to something a bit clearer, e.g `onScanComplete`

It's a standard to name UI callbacks `onSomething`, like `onChange` or `onKeyPressed`
So it makes it easier for devs to understand that you expect a callback